### PR TITLE
Sliding sync based on the nixpkgs module

### PR DESCRIPTION
**Pull request by D4ndellion**
It adds a couple of extra options to the nixpkgs sliding sync module but is otherwise fairly similar to #4

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -36,3 +36,19 @@ With matrix.YOURDOMAIN pointing at the server:
...
+```
+services.matrix-synapse-next = {
+  enableSlidingSync = true;
+};
+
+matrix-synapse.sliding-sync.environmentFile = "/some/file/containing/SYNCV3_SECRET=<some secret>";
+
+```
+
+If using well-known delagation make sure `YOURDOMAIN/.well-known/matrix/client` matches
```
Comment (h7x4) [inline README.MD]:
```suggestion
If using [well-known delagation](https://matrix-org.github.io/synapse/v1.37/delegate.html) make sure `YOURDOMAIN/.well-known/matrix/client` matches
```

File: sliding-sync/default.nix [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  options.services.matrix-synapse.sliding-sync = {
+    enable = lib.mkEnableOption (lib.mdDoc "sliding sync");
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.callPackage ../pkgs/matrix-sliding-sync { };
+      description = "What package to use for the sliding-sync proxy.";
+    };
+
+    enableNginx = lib.mkEnableOption "Should this module autogenerate nginx config";
```
Comment (h7x4) [inline sliding-sync/default.nix]: Docstring will be *Whether to enable Should this module autogenerate nginx config*

```suggestion
    enableNginx = lib.mkEnableOption lib.mdDoc("autogenerated nginx config");
```

File: synapse-module/default.nix [hunk 3]
```diff
@@ -84,6 +84,8 @@ in
       description = "A yaml python logging config file";
     };
 
+    enableSlidingSync = mkEnableOption "Automatic Sliding Sync setup at slidingsync.<domain>";
```
Comment (h7x4) [inline synapse-module/default.nix]:
```suggestion
    enableSlidingSync = mkEnableOption (lib.mdDoc "automatic Sliding Sync setup at `slidingsync.<domain>`");
```

File: synapse-module/default.nix [hunk 4]
```diff
@@ -394,5 +396,19 @@ in
         };
       };
     };
+
+    services.matrix-synapse-next.settings.extra_well_known_client_content."org.matrix.msc3575.proxy" = mkIf cfg.enableSlidingSync {
+      url = "https://${config.services.matrix-synapse.sliding-sync.publicBaseUrl}";
+    };
+    services.matrix-synapse.sliding-sync = mkIf cfg.enableSlidingSync {
+      enable = true;
+      enableNginx = cfg.enableNginx;
```
Comment (h7x4) [inline synapse-module/default.nix]: Is this option even needed if we're just going to blatantly set it equal to the main module option?

```suggestion
      enableNginx = lib.mkDefault cfg.enableNginx;
```

File: README.MD [hunk 5]
```diff
@@ -36,3 +36,19 @@ With matrix.YOURDOMAIN pointing at the server:
 ```
 
 is ~enough to get a functional matrix-server running with some workers
+
+## Sliding Sync (ElementX)
```
Comment (h7x4) [inline README.MD]:
```suggestion
## Sliding Sync (Element X)
```

File: README.MD [hunk 6]
```diff
@@ -36,3 +36,19 @@ With matrix.YOURDOMAIN pointing at the server:
...
+
+Just add the following to your config and point slidingsync.YOURDOMAIN at the server
```
Comment (h7x4) [inline README.MD]:
```suggestion
Just add the following to your config and point `slidingsync.YOURDOMAIN` at the server
```

**Review by h7x4 — CHANGES_REQUESTED** [review]
Looks pretty good, just a few suggestions

File: synapse-module/default.nix (hunk 4 above)
Comment (D4ndellion) [reply synapse-module/default.nix]: Yeah, the sliding sync module is completely disjoint from the synapse module, and could be exposed by its own

**Review by h7x4 — APPROVED** [review]

**Pull request merged**